src/main/java/io/jenkins/plugins/pipelinegraphview/utils/PipelineGraphApi.java

```diff
@@ -23,28 +25,57 @@
 
 public class PipelineGraphApi {
     private static final Logger logger = LoggerFactory.getLogger(PipelineGraphApi.class);
+    private static final BuildScheduleResult NOT_SCHEDULED =
+            new BuildScheduleResult.NotScheduled(Messages.scheduled_failure());
+
     private final transient WorkflowRun run;
 
     public PipelineGraphApi(WorkflowRun run) {
         this.run = run;
     }
 
     public Integer replay() throws ExecutionException, InterruptedException, TimeoutException {
-        run.checkPermission(Item.BUILD);
+        BuildScheduleResult result = scheduleBuild(run -> {
+            CauseAction causeAction = new CauseAction(new Cause.UserIdCause());
+            return Queue.getInstance()
+                    .schedule2(run.getParent(), 0, causeAction)
+                    .getItem();
+        });
+        // when java 21+ we can use switch expression
+        if (result instanceof BuildScheduleResult.Scheduled s) {
+            return s.buildNumber();
+        }
+        return null;
+    }
 
-        CauseAction causeAction = new CauseAction(new Cause.UserIdCause());
+    public @NonNull BuildScheduleResult scheduleBuild(Function<WorkflowRun, Queue.Item> scheduler) {
+        if (run == null) {
+            return NOT_SCHEDULED;
+        }
+
+        run.checkPermission(Item.BUILD);
 
         if (!run.getParent().isBuildable()) {
-            return null;
+            return NOT_SCHEDULED;
         }
 
-        Queue.Item item =
-                Queue.getInstance().schedule2(run.getParent(), 0, causeAction).getItem();
+        Queue.Item item = scheduler.apply(run);
+
         if (item == null) {
-            return null;
+            return NOT_SCHEDULED;
         }
 
-        return run.getParent().getNextBuildNumber();
+        return new BuildScheduleResult.Scheduled(run.getParent().getNextBuildNumber());
```

Comment:
I worry that this is just a guess and may not actually be correct but maybe that is fine?

Reply:
Further testing proves this isn't the right approach, just got a scenario where the number was one more than expected

Reply:
Maybe I should just instead set up a job to start polling once the build has been kicked off and once a next build is not null it can update the next build link and maybe generate a notification